nucypher/blockchain/eth/actors.py

```diff
@@ -666,6 +515,127 @@ def set_min_reward_rate_range(self,
         return receipt
 
 
+class Allocator:
```

Comment:
:boom:

Reply:
No inheritance from `BaseTokenActor` here which I'm guessing is because you dont want the superclass init logic?

Reply:
It's because I consider the `Allocator` a mere assistant to the `ContractAdministrator`. It's just a way to organize allocation logic in a single place.